examples/texture-arrays/main.rs

```diff
@@ -0,0 +1,323 @@
+            }),
+            rasterization_state: Some(wgpu::RasterizationStateDescriptor {
+                front_face: wgpu::FrontFace::Ccw,
+                cull_mode: wgpu::CullMode::None,
+                depth_bias: 0,
+                depth_bias_slope_scale: 0.0,
+                depth_bias_clamp: 0.0,
+            }),
+            primitive_topology: wgpu::PrimitiveTopology::TriangleList,
+            color_states: &[wgpu::ColorStateDescriptor {
+                format: sc_desc.format,
+                color_blend: wgpu::BlendDescriptor::REPLACE,
+                alpha_blend: wgpu::BlendDescriptor::REPLACE,
+                write_mask: wgpu::ColorWrite::ALL,
+            }],
+            depth_stencil_state: None,
+            vertex_state: wgpu::VertexStateDescriptor {
+                index_format: wgpu::IndexFormat::Uint16,
+                vertex_buffers: &[wgpu::VertexBufferDescriptor {
+                    stride: vertex_size as wgpu::BufferAddress,
+                    step_mode: wgpu::InputStepMode::Vertex,
+                    attributes: &wgpu::vertex_attr_array![0 => Float2, 1 => Float2, 2 => Int],
+                }],
+            },
+            sample_count: 1,
+            sample_mask: !0,
+            alpha_to_coverage_enabled: false,
+        });
+
+        (
+            Self {
+                vertex_buffer,
+                index_buffer,
+                bind_group,
+                pipeline,
+            },
+            None,
+        )
+    }
+    fn resize(
+        &mut self,
+        _sc_desc: &wgpu::SwapChainDescriptor,
+        _device: &wgpu::Device,
+        _queue: &wgpu::Queue,
+    ) {
+        // noop
+    }
+    fn update(&mut self, _event: winit::event::WindowEvent) {
+        // noop
+    }
+    fn render(
+        &mut self,
+        frame: &wgpu::SwapChainTexture,
+        device: &wgpu::Device,
+        _queue: &wgpu::Queue,
+    ) -> wgpu::CommandBuffer {
+        let mut encoder = device.create_command_encoder(&wgpu::CommandEncoderDescriptor {
+            label: Some("primary"),
+        });
+
+        let mut rpass = encoder.begin_render_pass(&wgpu::RenderPassDescriptor {
+            color_attachments: &[wgpu::RenderPassColorAttachmentDescriptor {
+                attachment: &frame.view,
+                resolve_target: None,
+                load_op: wgpu::LoadOp::Clear,
+                store_op: wgpu::StoreOp::Store,
+                clear_color: wgpu::Color {
+                    r: 0.0,
+                    g: 0.0,
+                    b: 0.0,
+                    a: 1.0,
+                },
+            }],
+            depth_stencil_attachment: None,
+        });
+        rpass.set_pipeline(&self.pipeline);
+        rpass.set_bind_group(0, &self.bind_group, &[]);
+        rpass.set_vertex_buffer(0, self.vertex_buffer.slice(..));
+        rpass.set_index_buffer(self.index_buffer.slice(..));
+        rpass.draw_indexed(0..12, 0, 0..1);
+
+        drop(rpass);
```

Comment:
I guess Rust can do this automatically for us, does it?

Reply:
It could if I put it in a block, but I much prefer the style where places where we need a drop are explicitly annotated to show "this object definitively ends here". Up to choice, but I like this style.

Reply:
no, I mean, even if you don't put it into a block and just remove the `drop`, will the compiler figure out that it has to drop the pass (since otherwise the `encoder` is borrowed and can't issue the `finish()` call)?

Reply:
Nope (just tried it). Lifetime data never influences where drops happen, so if something impls drop, it's going to have a nice `&mut self` at the end of the function, which ofc wouldn't work here.